pkg/gadgets/run/tracer/tracer.go

```diff
@@ -1085,6 +1106,35 @@ func (t *Tracer) runToppers(gadgetCtx gadgets.GadgetContext) {
 	}
 }
 
+func (t *Tracer) collectProfiler(gadgetCtx gadgets.GadgetContext) error {
+	cb := t.processEventFunc(gadgetCtx)
+
+	events := []*types.Event{}
+
+	key := make([]byte, t.histMap.KeySize(), t.histMap.KeySize())
+	value := make([]byte, t.histMap.ValueSize(), t.histMap.ValueSize())
+
+	it := t.histMap.Iterate()
+	for it.Next(&key, &value) {
+		event := cb(value)
+		events = append(events, event)
+
+		hist := &histogram.Histogram{
+			Intervals: histogram.NewIntervalsFromExp2Slots((*(*[]uint32)(unsafe.Pointer(&event.Blob[0])))[:t.histMap.ValueSize()>>2]),
+		}
+		fmt.Printf("%v\n", hist)
```

Comment:
Do you already know how to do this properly, or are you waiting for #2512 to be merged?